### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by TahenyBELGUITH**
In this milestone I : 
- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Add authentication to my API endpoints.
- my API endpoints should receive JSON and respond to JSON as well.

a code reviewer is appreciated  😄 ,
@TahenyBELGUITH

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  devise_for :users
+
+  devise_for :users
```
Comment (mmsesay) [inline config/routes.rb:5]: - [ ] I didn't see any API endpoint for authentication, both registration and login
       1.  Login endpoint - `POST : /login`
        2. Registration endpoint - `POST : /register or /signup`
       
For that reason, it was a challenge for me to test that functionality on your project. Can you please take a look at it? Thanks in advance.

File: config/routes.rb [hunk 2]
```diff
@@ -15,4 +16,8 @@
   end
     # devise_for :users, controllers: { confirmations: 'users/confirmations' }
   root to: 'users#index'
+  get 'api/user/:id/posts' => 'users#api_user_post', format: 'json'
+  get 'api/post/:id/comments' => 'posts#post_comments_api', format: 'json'  
+  post 'api/post/:id/comments' => 'comments#post_create_api'
```
Comment (mmsesay) [inline config/routes.rb]: - [ ] I was unable to add a comment for a post via the API. I got back `500 internal server error`
Please spare some time to take a look at it. I will be glad to know if it works well on your end.

Error 1 Screenshot |
:-----------------------:|
Screen Shot 2022-11-11 at 8 22 55 PM |



Error 2 Screenshot |
:--------------------------:|
Screen Shot 2022-11-11 at 8 23 09 PM |

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @TahenyBELGUITH, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 - Good Readme :heavy_check_mark: 
 - No linter error :heavy_check_mark: 
 - Good PR title :heavy_check_mark: 
 - Good PR summary :heavy_check_mark:  


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (TahenyBELGUITH) [reply config/routes.rb:5]: Hello mr @mmsesay,
thank you so much for your time , I see that you mention that i didn't put any endpoints for registration and login! 
but i did ... as you can see in the Screenshot 
Screenshot 2022-11-11 220310

Comment (mmsesay) [reply config/routes.rb:5]: Hi @TahenyBELGUITH thanks for the reply. 
From the screenshot, only the endpoints to `get` and `create a comment for a specific post` are shown.

**Review by FrederickMih — APPROVED** [review]
Hi @team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- PR is descriptive :heavy_check_mark: 
- No Linter error :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**